### PR TITLE
fix #32726, regression in subtyping `Tuple{Type{T},Vararg{T}}`

### LINKED ISSUE #32726: Unexpected MethodError on julia 1.3
When testing the julia 1.3 release candidate, we ran into the following issue.

Dispatch (sometimes) fails to match a valid method w/ subtyped type parameters. The following output shows the issue on julia 1.3:
```
julia> VERSION
v"1.3.0-DEV.554"

julia> f(::Type{T}, ::T...) where T = T
f (generic function with 1 method)

julia> f(Any, 1, 2.0, "hi")
ERROR: MethodError: no method matching f(::Type{Any}, ::Int64, ::Float64, ::String)
Closest candidates are:
  f(::Type{T}, ::T...) where T at REPL[2]:1
Stacktrace:
 [1] top-level scope at REPL[3]:1

julia> f(Any, 1, 2.0)
Any

julia> f(Any, 1, 2.0, "hi")
Any
```

The expected output (on v1.1.1):
```
julia> VERSION
v"1.1.1"

julia> f(::Type{T}, ::T...) where T = T
f (generic function with 1 method)

julia> f(Any, 1, 2.0, "hi")
Any

julia> f(Any, 1, 2.0)
Any

julia> f(Any, 1, 2.0, "hi")
Any
```

### PR DESCRIPTION
**Pull request by JeffBezanson**

### CODE REVIEW

**Comment by JeffBezanson** [comment]
linux32 failure looks like maybe an un-rooted type in codegen?

File: src/subtype.c [hunk 1]
```diff
@@ -1663,6 +1679,8 @@ JL_DLLEXPORT int jl_obvious_subtype(jl_value_t *x, jl_value_t *y, int *subtype)
             if (i < nparams_expanded_x) {
                 // there are elements left in x (possibly just a Vararg), check them against the Vararg tail of y too
                 assert(vy != JL_VARARG_NONE && istuple && iscov);
+                if (vararg_elt_bound(jl_tparam(y, i)))
+                    return 0;
```
Comment (vtjnash) [inline src/subtype.c]: Isn’t this just a rather complicated way of recomputing vt and then always returning false from here, or am I forgetting how normalization works now?

Comment (JeffBezanson) [reply src/subtype.c]: Ah, you're right. I guess instead we should check whether T in `Vararg{T}` is a typevar and whether it occurs invariant?

Comment (vtjnash) [reply src/subtype.c]: yeah, I guess we need to store a reference to the outermost type object so we could walk that here and look for uses inside invariant contexts (disabling some of the exactness tests below, but keeping the subtyping ones)?

File: src/subtype.c [hunk 2]
```diff
@@ -1665,6 +1665,8 @@ JL_DLLEXPORT int jl_obvious_subtype(jl_value_t *x, jl_value_t *y, int *subtype)
                 assert(vy != JL_VARARG_NONE && istuple && iscov);
                 jl_value_t *a1 = (vx != JL_VARARG_NONE && i >= npx - 1) ? vxt : jl_tparam(x, i);
                 jl_value_t *b = jl_unwrap_vararg(jl_tparam(y, i));
+                if (jl_is_typevar(b) && var_occurs_inside(y, (jl_tvar_t*)b, 0, 1))
```
Comment (vtjnash) [inline src/subtype.c]: Isn't this equivalent to:
```suggestion
                 if (jl_is_typevar(b))
```
Which seems not what you meant to write (though valid).

Comment (vtjnash) [reply src/subtype.c]: Or it's maybe actually:
```suggestion
                if (jl_is_typevar(b) && var_occurs_inside(jl_tparam(y, i), (jl_tvar_t*)b, 0, 1))
```
(e.g. it's also checking that the Vararg is not wrapped in a UnionAll with `b` as the variable—which seems reasonable to do, but should never appear in practice due to normalization, so again, equivalent to the above)

**Pull request merged**